src/index.js

```diff
@@ -1,27 +1,18 @@
 import { initialize } from "./cluster.js"
 import { getMongoConnection, getPostgresConnection } from './db.js'
 import cliProgress from 'cli-progress'
-import { setTimeout } from 'node:timers/promises'
+import os from 'os';
+
 const mongoDB = await getMongoConnection()
 const postgresDB = await getPostgresConnection()
-const ITEMS_PER_PAGE = 4000
-const CLUSTER_SIZE = 99
+// const ITEMS_PER_PAGE = 4000
+const CLUSTER_SIZE = os.cpus().length
```

Comment:
Using the CPU number will decrease the capacity of processing data in parallel. This would make sense if the child processes were relying on CPU intensive operations (here they're relying on Async I/O) that's why I have a big number of CPs running concurrently.

If you need to use CPU intensive ops such as crypto, processing hashes, images and etc, you'd use Worker Threads instead.

Thanks for the ideas anyway!!

Reply:
The ideal CLUSTER_SIZE depends on machine configurations and operations performed, making it a customizable setting. In this case, I set the number of clusters according to the number of CPUs just to provide a generic utilization for each type of PC configuration, which already yielded ideal results for this solution.